Zero size arrays are an extension

I see the following warning when compiling libgit2:

```
src/refs.h:66:12: warning: zero size arrays are an extension [-Wzero-length-array]
        char name[0];
                  ^
```

libgit2 homepage says it is portable and it should work with many compilers. Should the use of GCC extension be permitted here?

--- a/src/refs.h
+++ b/src/refs.h
@@ -63,7 +63,7 @@ struct git_reference {
 	} target;
 
 	git_oid peel;
-	char name[0];
+	char name[GIT_FLEX_ARRAY];
 };
 
 git_reference *git_reference__set_name(git_reference *ref, const char *name);

